src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -61,23 +97,24 @@ public partial class SourceArchive // Partial required for serialization
 
         public partial record SourceFileInfo(
             // Note: Path is also used as the key for source file retrieval
-            string Path,        // the location, relative to the main.bicep file's folder, for the file that will be shown to the end user (required in all Bicep versions)
-            string ArchivePath, // the location (relative to root) of where the file is stored in the archive (munged from Path, e.g. in case Path starts with "../")
-            string Kind,         // kind of source
-            string Contents
+            string Path,        // The location, relative to the main.bicep file's folder or one of the other roots.
+
+            string ArchivePath, // The location (relative to root) of where the file is stored in the archive (munged from Path, e.g. in case Path starts with "../")
+            string Kind,        // Kind of source
+            string Contents     // File contents
         );
 
         [JsonSerializable(typeof(ArchiveMetadata))]
         [JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
         private partial class MetadataSerializationContext : JsonSerializerContext { }
 
-        // Metadata for the entire archive (stored in __metadata.json file in archive)
+        // Metadata for the entire archive (stored in metadata.json file in archive)
         [JsonSerializable(typeof(ArchiveMetadata))]
         private record ArchiveMetadata(
-            int MetadataVersion,
+            int? MetadataVersion,
```

Comment:
Why not be strict about requiring this field?

Reply:
Makes sense.

Reply:
022fe99c2